### ISSUE TITLE
ClinVar XML

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#852]
We currently load ClinVar from separate GRCh37/38 VCFs

We should switch to the XML as it contains more variants and more info.

For analysis performance reasons we probably need it to continue to go into its own partition per build, and have a 1-to-1 to Variant model that summarises things

**lab specific classifications**

You can get stars but at the moment you can't see which labs submitted (some are more trustworthy than others) - we would like to bring in this info. Maybe it'll go into JSON

**CNVs**

To Find CNVs in ClinVar, easiest way is to search, then refine search by type or size

> ClinVar’s VCF file includes variants that are simple alleles (not a haplotype or genotype) < 10 kb in length with precise endpoints mapped to the GRCh37 or GRCh38 human genome assemblies. Other variants are not in scope, including cytogenetic variants, copy number variants with inner and/or outer start and stop coordinates, and variants >10 kb.

Here is a 1kb one: NC_000013.10:g.32889619_32890666dup  that is in VCF
[5kb](https://www.ncbi.nlm.nih.gov/clinvar/variation/1755741/) - NC_000013.11:g.32343971_32349243dup - in VCF after nov 2022 (not in vg.com as that's a bit old)
[16kb](https://www.ncbi.nlm.nih.gov/clinvar/variation/267628/) NC_000013.11:g.32319209_32335388dup not in VCF

[Download latest ClinVar XML](https://ftp.ncbi.nlm.nih.gov/pub/clinvar/xml/ClinVarFullRelease_00-latest.xml.gz)

**Comment by davmlaw** [comment]
@TheMadBug - are you able to paste your ClinVar XML code to this issue? Thanks

**Comment by TheMadBug** [comment]
[clinvar_parser.zip](https://github.com/SACGF/variantgrid/files/12158581/clinvar_parser.zip)

see attached - this was originally developed to change Clinvar's giant XML into manageable CSV files (500 rows each) for importing as test data into Shariant Test.

**Comment by davmlaw** [comment]
Before we meet - I think I need to actually look at the XML and see what's in there

1/2 way through something - how about zoom Monday?

**Comment by TheMadBug** [comment]
So the XML is the same XML Shariant submits to ClinVar - so it does contain the lab, the reviewer status etc.
For testing when I convert the XML to Classifications, I do ignore a lot of it, but you can see many many sample classifications in Shariant Test e.g.
https://test.shariant.org.au/classification/classification/104670

To my knowledge it is record by record, with no summary data.

**Comment by davmlaw** [comment]
TODO:

* Get a count of the types of novel variants in XML - big indels, uncertain ends, ISCN, etc

**Minimal CNV support**

* Pull out big indels and convert to a VCF (ie with extra CNV INFO fields)
* Write management command that "tacks on" this into ClinVarVersion

**Future work - put off for now**

I don't think it's worth changing everything and switching to the XML file now - can do it after SA Path upgrade

ClinVar are changing their XML format in fall of 2023 to better share somatic info - [see preview info](https://github.com/ncbi/clinvar)

VariantGrid currently loads ClinVar against Variant - this was historical as we started off with 1 build, and used VCFs (which are per-build). If we go to XML we could potentially change to link against Allele as that would allow us to only have 1. We already handle some stuff like this in Analysis (eg classifications and variant tags - there is only a slight performance hit vs going via variant)

We could potentially store all of the submissions/classifications in related objects, which would be useful for showing people on the variant page, classifications and generating discordances with ClinVar

Best to create a whole new model, and use the new postgres partitions - can load in parallel with old ones then switch with config to allow testing before deleting old ones.

ClinVar contains both builds (and clingen ID) we could use it for liftover during import by writing both builds linking to allele (liftover type = ClinVar)

Biocommons HGVS [doesn’t support uncertain offsets](https://github.com/biocommons/hgvs/issues/331) - could implement this

ISCN support? Have people entered this in the search box by chance?  did I put this somewhere? maybe a google doc?

Few questions about XML:

* Can we get all the info from VCF (stars etc?)
* What extra info can we get - lab submitter is a big one, also breaking up the individual classifications so you can see how they map to different diseases (currently clinvar mushes them all together into fields)

**Comment by davmlaw** [comment]
https://ncbiinsights.ncbi.nlm.nih.gov/2023/09/12/clinvar-somatic-variants-changes-xml/

**Comment by davmlaw** [comment]
🤖 Written by Claude

James's lazy-fetch design has landed: `annotation/clinvar_fetch_request.py`, `annotation/clinvar_xml_parser_via_vcv.py` and `ClinVarRecordCollection` fetch per-submitter records on demand. Two things were not done: bulk XML loading, and CNVs over 10kb. If ClinVar CNV coverage is still wanted after AnnotSV (see #1028), a narrow "ClinVar CNVs >10kb" issue would track it better than this one.
